patches/minecraft/net/minecraft/enchantment/EnchantmentHelper.java.patch

```diff
@@ -1,6 +1,201 @@
 --- ../src-base/minecraft/net/minecraft/enchantment/EnchantmentHelper.java
 +++ ../src-work/minecraft/net/minecraft/enchantment/EnchantmentHelper.java
-@@ -302,7 +302,7 @@
+@@ -30,6 +30,12 @@
+     private static final EnchantmentHelper.HurtIterator field_151388_d = new EnchantmentHelper.HurtIterator();
+     private static final EnchantmentHelper.DamageIterator field_151389_e = new EnchantmentHelper.DamageIterator();
+ 
++    public static int getEnchantmentLevel(Enchantment enchID, ItemStack stack, EntityLivingBase entityLivingBase)
++    {
++        return net.minecraftforge.common.ForgeHooks.getEnchantmentLevel(entityLivingBase, enchID, stack, func_77506_a(enchID, stack));
++    }
++
++    @Deprecated
+     public static int func_77506_a(Enchantment p_77506_0_, ItemStack p_77506_1_)
+     {
+         if (p_77506_1_.func_190926_b())
+@@ -108,11 +114,11 @@
+         }
+     }
+ 
+-    private static void func_77518_a(EnchantmentHelper.IModifier p_77518_0_, ItemStack p_77518_1_)
++    private static void applyEnchantmentModifier(EnchantmentHelper.IModifier modifier, ItemStack stack, @javax.annotation.Nullable EntityLivingBase entityLivingBase)
```

Comment:
The patches to this file have made the diff very strange and messy. Would it be better to make a new file like `EnchantmentHelper` in Forge, since you're already patching everything to use your new methods?

Reply:
It might indeed make sense to do this. As this would result in some code duplication, my only concern however is that this might make handling future Vanilla changes to those parts a bit more tricky to handle.